### PR TITLE
make fftshift and ifftshift take iterables as dimension arguments

### PR DESCRIPTION
**Pull request by JKrehl**
reincarnation of #16229

### CODE REVIEW

File: test/dsp.jl [hunk 1]
```diff
@@ -26,13 +26,25 @@ si = [0.9967207836936347,-1.4940914728163142,1.2841226760316475,-0.4524417279474
 @test_throws ArgumentError filt!([1, 2], [1], [1], [1])
 @test xcorr([1, 2], [3, 4]) == [4, 11, 6]
 
+# Shift-Functions
 @test fftshift([1 2 3]) == [3 1 2]
 @test fftshift([1, 2, 3]) == [3, 1, 2]
 @test fftshift([1 2 3; 4 5 6]) == [6 4 5; 3 1 2]
+
+￼@test fftshift([1 2 3; 4 5 6], 1) == [4 5 6; 1 2 3]
```
Comment (tkelman) [inline test/dsp.jl]: what's with the `￼` ?

File: test/dsp.jl [hunk 2]
```diff
@@ -26,13 +26,25 @@ si = [0.9967207836936347,-1.4940914728163142,1.2841226760316475,-0.4524417279474
...
+@test fftshift([1 2 3; 4 5 6], 1:2) == [6 4 5; 3 1 2]
+
 @test ifftshift([1 2 3]) == [2 3 1]
 @test ifftshift([1, 2, 3]) == [2, 3, 1]
 @test ifftshift([1 2 3; 4 5 6]) == [5 6 4; 2 3 1]
 
+@test ifftshift([1 2 3; 4 5 6], 1) == [4 5 6; 1 2 3]
+￼@test ifftshift([1 2 3; 4 5 6], ()) == [1 2 3; 4 5 6]
+￼@test ifftshift([1 2 3; 4 5 6], (1,2)) == [5 6 4; 2 3 1]
+￼@test ifftshift([1 2 3; 4 5 6], 1:2) == [5 6 4; 2 3 1]
```
Comment (tkelman) [inline test/dsp.jl]: 3 more of them here

Comment (JKrehl) [inline test/dsp.jl]: I only found one. But as Travis seems to be faster than my machine...

Comment (tkelman) [inline test/dsp.jl]: lines 45 and 46 too

**Comment by StefanKarpinski** [comment]
CI passes – @stevengj, please have a quick look at this again and merge at will.

**Pull request merged**